nanoFramework.System.Threading/System/Runtime/CompilerServices/AsyncTaskMethodBuilder.cs

```diff
@@ -9,6 +9,108 @@
+        /// </summary>
+        /// <returns>A new instance of the builder.</returns>
+        public static AsyncTaskMethodBuilder Create()
+        {
+            Debug.WriteLine($"AsyncTaskMethodBuilder:Create");
+            return new AsyncTaskMethodBuilder();
+        }
+
+        Task task;
+        /// <summary>
+        /// Gets the task for this builder
+        /// </summary>
+        /// <value>
+        /// The task for this builder.
+        /// </value>
+        public Task Task => task;
+
+        /// <summary>
+        ///  Schedules the state machine to proceed to the next action when the specified awaiter completes.
+        /// </summary>
+        /// <typeparam name="TAwaiter"></typeparam>
+        /// <typeparam name="TStateMachine"></typeparam>
+        /// <param name="awaiter"></param>
+        /// <param name="stateMachine"></param>
+        public void AwaitOnCompleted<TAwaiter, TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine)
+            where TAwaiter : INotifyCompletion
+            where TStateMachine : IAsyncStateMachine
+        {
+            Debug.WriteLine($"AsyncTaskMethodBuilder:AwaitOnCompleted");
+            var _stateMachine = stateMachine;
+            awaiter.OnCompleted(() =>
+            {
+                Debug.WriteLine($"AsyncTaskMethodBuilder:OnCompleted");
+                _stateMachine.MoveNext();
+            });
+        }
+
+        /// <summary>
+        /// Schedules the state machine to proceed to the next action when the specified awaiter completes. This method can be called from partially trusted code.
+        /// </summary>
+        /// <typeparam name="TAwaiter"></typeparam>
+        /// <typeparam name="TStateMachine"></typeparam>
+        /// <param name="awaiter"></param>
+        /// <param name="stateMachine"></param>
+        public void AwaitUnsafeOnCompleted<TAwaiter, TStateMachine>(ref TAwaiter awaiter, ref TStateMachine stateMachine)
+            where TAwaiter : ICriticalNotifyCompletion
+            where TStateMachine : IAsyncStateMachine
+        {
+            Debug.WriteLine($"AsyncTaskMethodBuilder:AwaitUnsafeOnCompleted");
+            var _stateMachine = stateMachine;
+            awaiter.OnCompleted(() =>
+            {
+                Debug.WriteLine($"AsyncTaskMethodBuilder:OnCompleted");
+                _stateMachine.MoveNext();
+            });
+        }
+
+        /// <summary>
+        ///  Marks the task as failed and binds the specified exception to the task.
+        /// </summary>
+        /// <param name="exception"></param>
+        public void SetException(Exception exception)
+        {
+            Debug.WriteLine($"AsyncTaskMethodBuilder:SetException");
+            task?.CompleteWithException(exception);
+        }
+        /// <summary>
+        /// Marks the task as successfully completed.
+        /// </summary>
+        public void SetResult()
+        {
+            Debug.WriteLine($"AsyncTaskMethodBuilder:SetResult");
```

Comment:
Apart from the debug output nothing else is happening, is this the intend behaviour?